### PR TITLE
Create Vercel app for onefetch with ASCII preview

### LINKED ISSUE #696: Automated logo rendering/visualization
### Summary 💡

Started the conversation [here](https://github.com/o2sh/onefetch/issues/328#issuecomment-1172794745), but my comment was unrelated enough that it should have been a new issue.

Basic idea is that we could have a solution to automate previews of the logos. This can probably be done with a serverless function.

Some things to consider:
- What tool to use for the serverless function. I've used Vercel, but there are plenty of options out there.
- We'll need the languages' color definitions to be available to the serverless function. We can do this in a few ways:
  - Convert the language definitions to a data format (e.g. JSON) so that a serverless function can read it. If we do this we can
    - make a proc macro to parse the JSON in the Rust code
    - use a build binary to generate the Rust code
    - either way we probably wouldn't want to do something that results in runtime JSON parsing
  - Add another binary to the onefetch crate, that will be able to import the language definitions and act as the serverless function
  - Export the language definitions with a minimal library and use a different crate for the serverless function. If we do this, a couple more things to consider (this can be influenced by the serverless utility and how it reacts to repository changes):
    - Should the serverless function be part of a workspace, in the same repo?
    - ~~Should it exist in a different repo?~~ (we probably want it in the same repo to listen for changes and be able to import the latest change from the main onefetch project)
- Does the serverless function render the HTML? It may be more efficient to generate HTML once and serve a static site, but we can dynamically generate HTML per request.
- Do we simply serve HTML, or should we have an API for the language stats? If we have an API, we can use a reactive library to read JSON and generate a preview. The API could be a simple `api/language[/:name]`.

Also if we just generate the HTML once, we can actually just use GH Pages instead of a serverless tool. But one thing I've liked about Vercel is that it creates preview builds for PRs (see motivation).

### Motivation 🔦

When reviewing PRs, currently we have to pull down the PR's changes and rebuild the binary to review any added or changed logos. This can be greatly simplified by making previews that are automatically generated, which can allow maintainers to review the logo even if they're not at their main device.

### PR DESCRIPTION
**Pull request by spenserblack**
At the time of creation of this PR, there is absolutely nothing notable, hence why it is starting as a draft. I am creating this before any significant code is contributed so that I can document the steps as I go through them.

## Setup

1. Create a Vercel account at vercel.com and link it to GitHub
2. Create a new project
   - set repo to onefetch 
   - change the root directory from `./` to ~~`vercel/ascii-preview`~~ `vercel/`
   - set the framework preset to Vite
4. Change the domain in project settings(?)
   - My current URL is https://onefetch-spenserblack.vercel.app/

Resolves #696

### CODE REVIEW

**Comment by spenserblack** [comment]
I think I'll end up using TypeScript for this (using Vite as a build tool). Before I set up the preview, @o2sh do you have a preference between vanilla JS, React, Vue.js, Svelte, or anything else? I've used Vue.js and Svelte before (and obviously vanilla JS 😆), but I'm happy to use any tool that you and the team feel most comfortable with.

File: .github/workflows/ci.yml
Comment (o2sh) [inline .github/workflows/ci.yml:8]: This folder may need to be added to the `exclude` section of the `Cargo.toml` file.

**Comment by o2sh** [comment]
> Before I set up the preview, @o2sh do you have a preference between vanilla JS, React, Vue.js, Svelte, or anything else?

Why the need of a front end framework/library for a serverless function?

>Also, if we ever want other serverless onefetch web services, maybe it makes more sense to make the root directory vercel/ instead, and /ascii-preview can be an endpoint instead.

Aren't Vercel's serverless functions supposed to be  placed in the `/api` directory of the project's root. https://vercel.com/docs/concepts/functions/serverless-functions#deploying-serverless-functions

**Comment by spenserblack** [comment]
>Why the need of a front end framework/library for a serverless function?

For ASCII previews itself, no big reason, just that it might be a bit easier to parse data into a template this way. Also, reactively changing the background from dark to light can be useful. While not impossible without a frontend framework, it's a bit easier, and also more readable for other potential contributors. Though if we decide to use any framework now, this simplifies adding anything else to a serverless web service. For example, https://github.com/o2sh/onefetch/issues/459#issuecomment-874654998 was suggesting moving the wiki to gh-pages, but something very similar can be accomplished using Vercel instead. Though I have no strong feelings, and would be happy to implement this with Vanilla JS (no frameworks/libraries).

> Aren't Vercel's serverless functions supposed to be placed in the `/api` directory of the project's root.

Well, the serverless function don't *need* to be in `/api`. For example, [github-profile-trophy](https://github.com/ryo-ma/github-profile-trophy) is basically a single serverless function run with Deno that returns an SVG, and this serverless function exists in the root. E.g. https://github-profile-trophy.vercel.app/?username=o2sh (congrats on the achievements 😆 )

Personally, I'd put machine-readable responses in `/api[:/endpoint]`, and human-readable responses anywhere else. We can split this up into, let's say, `/api/languages` which serves a JSON-ified version of `languages.yaml`, and `/ascii-preview` that fetches and consumes that JSON.

I think I may have been incorrectly using the phrase "serverless function." I was using it as a catch-all for static sites as well. Sorry for the confusion.

**Comment by o2sh** [comment]
> While not impossible without a frontend framework, it's a bit easier, and also more readable for other potential contributors. Though if we decide to use any framework now, this simplifies adding anything else to a serverless web service. 

You're right about the need of being future proof.
I'm always anxious about bringing more complexity to a project. 😩 
Concerning your original question - which framework/library to use -, you can pick the one you feel most comfortable with and is most adapted to the feature. I have a small preference for (Svelte + TypeScript) 😉 .

> Personally, I'd put machine-readable responses in /api[:/endpoint], and human-readable responses anywhere else. We can split this up into, let's say, /api/languages which serves a JSON-ified version of languages.yaml, and /ascii-preview that fetches and consumes that JSON.

If we're adding folders to the project's root, we should keep it to a minimum. For example, a single folder named `tools` - or another name that better illustrate our intent - which can then hold as many folders as we consider necessary.

**Comment by spenserblack** [comment]
>If we're adding folders to the project's root, we should keep it to a minimum. For example, a single folder named tools - or another name that better illustrate our intent - which can then hold as many folders as we consider necessary.

Oh yeah, right now I'm planning on putting everything in `vercel/`, so `vercel/api/`, `vercel/index.html`, etc. I think that's what you're saying you want, right?

File: vercel/api/languages/[key].ts
Comment (o2sh) [inline vercel/api/languages/[key].ts:12]:
```suggestion
  response.status(200).send(language);
```

File: vercel/api/languages/[key].ts
Comment (o2sh) [inline vercel/api/languages/[key].ts:6]:
```suggestion
  const key = request.query.key as string;
```

File: vercel/tsconfig.json
Comment (o2sh) [inline vercel/tsconfig.json:2]: I had to remove this line for it to work locally, otherwise I would get this error when calling api/languages:

` TS5095: Option 'preserveValueImports' can only be used when 'module' is set to 'es2015' or later.`

File: vercel/src/lib/AsciiPreview.svelte
Comment (o2sh) [inline vercel/src/lib/AsciiPreview.svelte:62]:
```suggestion
    background-color: black;
```
Personal preference.

File: vercel/src/lib/AsciiPreview.svelte
Comment (o2sh) [inline vercel/src/lib/AsciiPreview.svelte:21]: We should add the terminal default foreground logic: `if white (or #FFFFFF) && !Dark --> Black`

File: vercel/src/lib/AsciiPreview.svelte
Comment (o2sh) [inline vercel/src/lib/AsciiPreview.svelte:32]:
```suggestion
<h3 style="background-color:{chip}; color:white">#{name}</h3>
```
Just a suggestion to use the chip color.

**Comment by o2sh** [comment]
This is really cool @spenserblack 😍
I am still working on the review. But it's already looking very promising.

File: vercel/tsconfig.json
Comment (spenserblack) [reply vercel/tsconfig.json:2]: :thinking: That's weird, it's part of the Vercel default, so I'd assume it would be compatible with Vercel serverless functions in Typescript.
:shrug: Well, as long as its deployment is successful 😆

File: vercel/src/lib/AsciiPreview.svelte
Comment (spenserblack) [reply vercel/src/lib/AsciiPreview.svelte:32]: Interesting idea! I'll play around with this a bit.

Comment (spenserblack) [reply vercel/src/lib/AsciiPreview.svelte]: I decided to make the chip color a circle next to the language's title in https://github.com/o2sh/onefetch/pull/701/commits/ad93cdf215cbe0151449edea3eaecaf3fd3757c5. I was concerned about readability by coloring the title itself.

File: vercel/src/lib/AsciiPreview.svelte
Comment (o2sh) [inline vercel/src/lib/AsciiPreview.svelte:34]: minor: <h3> should not be a container

```suggestion
   <div class="name-with-chip">
    <Chip color={chip} width={24} height={24} />
    <h3>{name}</h3>
  </div>
```

css:
```
.name-with-chip {
    display: flex;
    align-items: baseline;
    gap: 1.5rem;
  }
```

File: vercel/src/lib/AsciiPreview.svelte
Comment (o2sh) [reply vercel/src/lib/AsciiPreview.svelte]: That's way better👍

File: vercel/src/ascii-preview/Index.svelte
Comment (o2sh) [inline vercel/src/ascii-preview/Index.svelte:29]:
```suggestion
  {#await languages}
    <p>Fetching languages...</p>
  {:then result}
    <h2>Languages ({result.length})</h2>
```

In order to display the number of languages

Comment (spenserblack) [reply vercel/src/ascii-preview/Index.svelte:29]: What do you think about `{{ result.length }} Languages`? Think it's bit more like plain English.

Comment (o2sh) [reply vercel/src/ascii-preview/Index.svelte:29]: Yes, `{result.length } languages` may be more readable

Comment (spenserblack) [reply vercel/src/ascii-preview/Index.svelte:29]: Oh, sorry, I completely misread your code suggestion and saw it as a Vue-like `{{ value }}`. I didn't realize there were parentheses 😆

File: vercel/src/lib/AsciiPreview.svelte
Comment (o2sh) [reply vercel/src/lib/AsciiPreview.svelte]: You may have missed the css part.

Comment (spenserblack) [reply vercel/src/lib/AsciiPreview.svelte]: You're right, I just added it back.

File: vercel/src/lib/Nav.svelte
Comment (o2sh) [inline vercel/src/lib/Nav.svelte:41]: minor: the bg color is barely noticeable,

File: vercel/src/ascii-preview/Index.svelte
Comment (o2sh) [inline vercel/src/ascii-preview/Index.svelte:23]: This dive can be removed ?

File: vercel/src/ascii-preview/Index.svelte
Comment (o2sh) [inline vercel/src/ascii-preview/Index.svelte:25]:
```suggestion
   <Nav active="/ascii-preview" />
```

File: vercel/src/lib/AsciiPreview.svelte
Comment (o2sh) [inline vercel/src/lib/AsciiPreview.svelte:23]:
```suggestion
          : `<span style="color: ${ansi[i]}">`;
```
I don't see the point of having an extra stylesheet for this (app.css).

Comment (spenserblack) [reply vercel/src/lib/AsciiPreview.svelte]: I guess I was just being a bit obsessive about the "use stylesheets instead of the style attribute" rule 😆

**Comment by o2sh** [comment]
@spenserblack I seem to have broken the serverless function 😓 

Do you have access to the logs ?

**Comment by spenserblack** [comment]
:thinking: Looks like the `/api/languages` endpoint broke

**Comment by spenserblack** [comment]
# Build log

```
[14:42:20.483] Cloning github.com/spenserblack/onefetch (Branch: main, Commit: 85c975a)
[14:42:20.887] Cloning completed: 403.888ms
[14:42:21.292] Installing build runtime...
[14:42:22.485] Build runtime installed: 1.192s
[14:42:22.518] Installing build runtime...
[14:42:23.477] Build runtime installed: 959.107ms
[14:42:23.506] Installing build runtime...
[14:42:26.518] Build runtime installed: 3.012s
[14:42:27.165] Looking up build cache...
[14:42:27.281] Looking up build cache...
[14:42:27.804] Looking up build cache...
[14:42:30.549] Build cache downloaded [37.30 MB]: 2931.452ms
[14:42:30.594] Build cache downloaded [37.30 MB]: 2934.876ms
[14:42:30.922] Build cache downloaded [37.30 MB]: 2818.664ms
[14:42:31.423] Installing dependencies...
[14:42:31.907] yarn install v1.22.17
[14:42:31.924] Installing dependencies...
[14:42:31.981] [1/4] Resolving packages...
[14:42:32.085] success Already up-to-date.
[14:42:32.089] Done in 0.19s.
[14:42:32.289] yarn install v1.22.17
[14:42:32.357] [1/4] Resolving packages...
[14:42:32.399] Running "yarn run build"
[14:42:32.426] Installing dependencies...
[14:42:32.452] success Already up-to-date.
[14:42:32.457] Done in 0.17s.
[14:42:32.638] yarn run v1.22.17
[14:42:32.686] $ yarn setup:data && yarn build:vite
[14:42:32.706] yarn install v1.22.17
[14:42:32.727] Using TypeScript 4.7.4 (local user-provided)
[14:42:32.793] [1/4] Resolving packages...
[14:42:32.932] success Already up-to-date.
[14:42:32.936] Done in 0.23s.
[14:42:32.983] $ js-yaml ../languages.yaml > languages.json
[14:42:33.231] Using TypeScript 4.7.4 (local user-provided)
[14:42:33.564] $ vite build
[14:42:34.197] [36mvite v3.0.0 [32mbuilding for production...[36m[39m
[14:42:34.247] transforming...
[14:42:35.555] [31mNowBuildError: api/languages/[key].ts(1,18): error TS2307: Cannot find module '../../languages.json' or its corresponding type declarations.
[14:42:35.555] 
[14:42:35.555]     at createTSError (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:305069:16)
[14:42:35.555]     at reportTSError (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:305075:23)
[14:42:35.555]     at getOutputTypeCheck (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:305162:17)
[14:42:35.555]     at compile (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:305239:109)
[14:42:35.555]     at compileTypeScript (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:304587:31)
[14:42:35.555]     at Job.readFile (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:304614:30)
[14:42:35.555]     at Job.emitDependency (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:227618:39)
[14:42:35.555]     at async Promise.all (index 0)
[14:42:35.555]     at async Object.nodeFileTrace (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:227372:5)
[14:42:35.556]     at async compile (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:304596:49) {
[14:42:35.556]   hideStackTrace: true,
[14:42:35.556]   code: 'NODE_TYPESCRIPT_ERROR',
[14:42:35.556]   link: undefined,
[14:42:35.556]   action: undefined
[14:42:35.556] }[0m
[14:42:35.681] 6:42:35 PM [vite-plugin-svelte] /vercel/path3/vercel/src/lib/AsciiPreview.svelte:37:2 A11y: A form label must be associated with a control.
[14:42:35.682] 35:     Dark
[14:42:35.682] 36:   </label>
[14:42:35.682] 37:   <label>
[14:42:35.682]       ^
[14:42:35.683] 38:     {#if hex != null}
[14:42:35.683] 39:       <input type="checkbox" bind:checked={trueColor} />
[14:42:35.956] [32m✓[39m 20 modules transformed.
[14:42:36.016] rendering chunks...
[14:42:36.035] [90m[37m[2mdist/[22m[90m[39m[32mindex.html                        [39m [2m0.51 KiB[22m
[14:42:36.036] [90m[37m[2mdist/[22m[90m[39m[32mascii-preview/index.html          [39m [2m0.54 KiB[22m
[14:42:36.076] [90m[37m[2mdist/[22m[90m[39m[36massets/main.a2c986bb.js           [39m [2m1.20 KiB / gzip: 0.65 KiB[22m
[14:42:36.076] [90m[37m[2mdist/[22m[90m[39m[36massets/ascii-preview.5c75593a.js  [39m [2m5.31 KiB / gzip: 2.48 KiB[22m
[14:42:36.077] [90m[37m[2mdist/[22m[90m[39m[36massets/Nav.0fa23d3b.js            [39m [2m6.09 KiB / gzip: 2.75 KiB[22m
[14:42:36.077] [90m[37m[2mdist/[22m[90m[39m[35massets/index.bf0a7ac8.css         [39m [2m0.04 KiB / gzip: 0.06 KiB[22m
[14:42:36.077] [90m[37m[2mdist/[22m[90m[39m[35massets/index.9483747b.css         [39m [2m0.31 KiB / gzip: 0.19 KiB[22m
[14:42:36.077] [90m[37m[2mdist/[22m[90m[39m[35massets/Nav.5a5a1fc4.css           [39m [2m3.09 KiB / gzip: 1.05 KiB[22m
[14:42:36.078] Done in 3.43s.
[14:42:36.078] [31mNowBuildError: api/languages/index.ts(1,18): error TS2307: Cannot find module '../../languages.json' or its corresponding type declarations.
[14:42:36.078] 
[14:42:36.078]     at createTSError (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:305069:16)
[14:42:36.078]     at reportTSError (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:305075:23)
[14:42:36.078]     at getOutputTypeCheck (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:305162:17)
[14:42:36.078]     at compile (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:305239:109)
[14:42:36.079]     at compileTypeScript (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:304587:31)
[14:42:36.079]     at Job.readFile (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:304614:30)
[14:42:36.079]     at Job.emitDependency (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:227618:39)
[14:42:36.079]     at async Promise.all (index 0)
[14:42:36.079]     at async Object.nodeFileTrace (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:227372:5)
[14:42:36.079]     at async compile (/vercel/b1f299b72c92121c/.build-utils/.builder/node_modules/@vercel/node/dist/index.js:304596:49) {
[14:42:36.079]   hideStackTrace: true,
[14:42:36.079]   code: 'NODE_TYPESCRIPT_ERROR',
[14:42:36.080]   link: undefined,
[14:42:36.080]   action: undefined
[14:42:36.080] }[0m
[14:42:39.290] Generated build outputs:
[14:42:39.290]  - Static files: 10
[14:42:39.290]  - Serverless Functions: 2
[14:42:39.290]  - Edge Functions: 0
[14:42:39.290] Serverless regions: Washington, D.C., USA
[14:42:39.290] Deployed outputs in 3s
[14:42:41.832] Build completed. Populating build cache...
[14:42:46.671] Uploading build cache [37.30 MB]...
[14:42:48.020] Build cache uploaded: 1.350s
[14:42:48.048] Done with "package.json"
[14:42:49.039] Build completed. Populating build cache...
[14:42:53.469] Uploading build cache [37.30 MB]...
[14:42:54.831] Build cache uploaded: 1.362s
[14:42:54.856] Done with "api/languages/index.ts"
```

# Serverless Function Error

```
[GET] /api/languages
15:22:51:16
2022-07-22T19:22:51.333Z	undefined	ERROR	Cannot find module '../../languages.json'
Require stack:
- /var/task/vercel/api/languages/index.js
2022-07-22T19:22:51.333Z	undefined	ERROR	Did you forget to add it to "dependencies" in `package.json`?
RequestId: 36e7188d-882b-4fd7-ba3e-b4f1f2f30f12 Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

**Comment by spenserblack** [comment]
I think the issue might be that the API and site are built in separate processes. We may have just been lucky that the site build, which included `setup:data`, completed first.

Useful notes: https://vercel.com/docs/runtimes#advanced-usage/advanced-node-js-usage/custom-build-step-for-node-js

**Comment by o2sh** [comment]
That's weird, looking at the logs, `"build": "yarn setup:data && yarn build:vite",` seems to have run sequentially.

Seems like `language.json` wasn't found for some reason.

**Comment by o2sh** [comment]
with d30407a I got rid of the API and read directly from the json file to see if it works and better

It may simplify things after all. 

Feel free to revert if you think this is a bad idea.

**Comment by spenserblack** [comment]
>Feel free to revert if you think this is a bad idea.

Oh, no, my original idea was actually to import the YAML directly. I did this by making a symlink to `languages.yaml` in `vercel/`, but it failed on deployment. I assume this was because files outside of the deployed directory weren't available, but maybe the problem was just the symlink itself :shrug:

It makes the bundled JS a bit large, but I think the simplicity is worth it 🙂

**Review by spenserblack — COMMENTED** [review]
There's both a `package-lock.json` and a `yarn.lock` now. We should probably pick one or the other.

File: vercel/tsconfig.json
Comment (spenserblack) [inline vercel/tsconfig.json:2]: Since there aren't any type errors in the API anymore (since there is no API 😆 ), what do you think about adding back the `"extends": "@tsconfig/svelte/tsconfig.json"`?

File: vercel/src/lib/AsciiPreview.svelte
Comment (spenserblack) [inline vercel/src/lib/AsciiPreview.svelte:22]: Should we remap the true color? The onefetch binary doesn't remap true colors, and it might be useful to soo how the logo will render if it's white-on-white.

File: vercel/static/sakura.css
Comment (spenserblack) [inline vercel/static/sakura.css:1]: Might want to add a [`linguist-vendored`](https://github.com/github/linguist/blob/master/docs/overrides.md) to the `.gitattributes` for this fiel.

File: vercel/src/lib/AsciiPreview.svelte
Comment (o2sh) [reply vercel/src/lib/AsciiPreview.svelte:22]: You're right, I thought we did, my bad 🤦 
The whole point of the ascii preview is to be iso with the binary.

File: vercel/static/sakura.css
Comment (o2sh) [reply vercel/static/sakura.css:1]: Maybe we should exclude the vercel folder all together. I'm afraid of users being confused on whether the tool is actually `completely offline` - as stated in the README

Comment (spenserblack) [reply vercel/static/sakura.css:1]: Good point. I'm not sure if there's an effective difference, but maybe the website should be `linguist-documentation`?

Comment (o2sh) [reply vercel/static/sakura.css:1]: `linguist-documentation` seems more appropriate.

**Comment by o2sh** [comment]
TypeScript is really giving a hard time with this import 😓 

Screenshot 2022-07-27 at 23 04 20

seems like it doesn't want me to import a file that is one level up from its root path 🤔

**Comment by spenserblack** [comment]
>TypeScript is really giving me a hard time with this import

I had that issue too, which is why I caved in and made the YAML -> JSON a build step. TypeScript natively supports importing JSON, but the YAML import is handled by a Vite plugin, so I'm not sure if the TypeScript compiler is aware of that.

I think something like this might be necessary.
```typescript
// types.d.ts
declare module '*.yaml' {
  const data: Record<string, any>;
  export default data;
}
```

**Comment by spenserblack** [comment]
https://github.com/o2sh/onefetch/pull/701/commits/c93ef824509317be687eaad598f7ce70084cb540 Resolved the type errors for me (running `npm run check`).

**Comment by spenserblack** [comment]
:thinking: I wonder if running `npm run check` should be made part of the CI. `npm run build` would be asserted by Vercel, so that might not be necessary.

**Comment by spenserblack** [comment]
:cry: Looks like nightly had a change that broke the CI.

**Review by o2sh — APPROVED** [review]
I think the PR is ready to be merged 🎉 

Maybe we can wait for gitoxide to fix the build issue 🤔 [their CI is also failing](https://github.com/Byron/gitoxide/runs/7562576497?check_suite_focus=true)

File: vercel/public/sakura.css
Comment (spenserblack) [inline vercel/public/sakura.css:5]: Since this has been modified by us, we might want to add a comment here just saying so. We don't *need* to, but it might be helpful for someone stumbling on this.

**Comment by spenserblack** [comment]
>I think the PR is ready to be merged

I think so, too! I'm fine to wait until the CI is fixed. Thanks for all your work updating the styling, simplifying the logic, etc.!

File: vercel/src/lib/AsciiPreview.svelte
Comment (spenserblack) [inline vercel/src/lib/AsciiPreview.svelte:45]: Just a minor nitpick:

I just realized that, now that the `<input>` isn't wrapped in a `<label>`, clicking on the text "Dark" doesn't do anything (previously it would check/uncheck the box just like clicking on the input itself). If we can't use a wrapping `<label>`, perhaps `<label for="dark-checkbox-{index}">` + `<input id="dark-checkbox-{index}">` (this would accomplish the same behavior)?

Comment (spenserblack) [reply vercel/src/lib/AsciiPreview.svelte]: Done in https://github.com/o2sh/onefetch/pull/701/commits/558ccdbc0950502fd349b7905e4a96d63612187f and https://github.com/o2sh/onefetch/pull/701/commits/b64d590a67c6a13d683b72b8e81df22124432655

**Comment by spenserblack** [comment]
:thinking: Maybe we should add `vercel/` to `CODEOWNERS`.

File: vercel/src/lib/AsciiPreview.svelte
Comment (o2sh) [inline vercel/src/lib/AsciiPreview.svelte:46]: Isn't the `name` unique? maybe we could use it as a key.

Comment (o2sh) [reply vercel/src/lib/AsciiPreview.svelte]: done in [90fcea0](https://github.com/o2sh/onefetch/pull/701/commits/90fcea0986ef299f2e8cff29a9b03656133a7753)

**Comment by spenserblack** [comment]
Should onefetch.dev be listed as one of the [sites using sakura.css](https://github.com/oxalorg/sakura#sites-using-sakura)?

File: vercel/src/lib/AsciiPreview.svelte
Comment (spenserblack) [reply vercel/src/lib/AsciiPreview.svelte]: 😆 Good point, I had completely forgotten that the name is obviously unique.

**Comment by o2sh** [comment]
Sure! that's a good idea 👍

**Comment by spenserblack** [comment]
https://github.com/oxalorg/sakura/pull/97

**Pull request merged**
